openhands/server/routes/manage_conversations.py

```diff
@@ -244,16 +247,65 @@ async def get_conversation(
         return None
 
 
+async def auto_generate_title(conversation_id: str, user_id: str | None) -> str:
+    """
+    Auto-generate a title for a conversation based on the first user message.
+
+    Args:
+        conversation_id: The ID of the conversation
+        user_id: The ID of the user
+
+    Returns:
+        A generated title string
+    """
+    logger.info(f'Auto-generating title for conversation {conversation_id}')
+
+    try:
+        # Create an event stream for the conversation
+        event_stream = EventStream(conversation_id, file_store, user_id)
+
+        # Find the first user message
+        first_user_message = None
+        for event in event_stream.get_events():
+            if (
+                event.source == EventSource.USER
+                and isinstance(event, MessageAction)
+                and event.content
+                and event.content.strip()
+            ):
+                first_user_message = event.content
+                break
+
+        if first_user_message:
+            first_user_message = first_user_message.strip()
+            title = first_user_message[:15]
+            if len(first_user_message) > 15:
+                title += '...'
+            logger.info(f'Generated title: {title}')
+            return title
+    except Exception as e:
+        logger.error(f'Error generating title: {str(e)}')
+    return ''
+
+
 @app.patch('/conversations/{conversation_id}')
 async def update_conversation(
     request: Request, conversation_id: str, title: str = Body(embed=True)
 ) -> bool:
+    user_id = get_user_id(request)
     conversation_store = await ConversationStoreImpl.get_instance(
-        config, get_user_id(request), get_github_user_id(request)
+        config, user_id, get_github_user_id(request)
     )
     metadata = await conversation_store.get_metadata(conversation_id)
     if not metadata:
         return False
+
+    # If title is empty or unspecified, auto-generate it from the first user message
+    if not title or title.isspace():
+        title = await auto_generate_title(conversation_id, user_id)
```

Comment:
Actually i have a function that i've tested to work in another PR (but haven't got time to get it merged) that could fit well here - It can probably improve this once this got merged

https://github.com/All-Hands-AI/OpenHands/pull/7049/files#diff-bd7df3830070d1adcb411d3a60d3e56f6182014e7586422cbfc429c4dc6bfee6